src/test/javascript/spec/component/assessment-shared/assessment-locks.component.spec.ts

```diff
@@ -31,24 +55,64 @@ describe('AssessmentLocksComponent', () => {
                 { provide: LocalStorageService, useClass: MockSyncStorage },
                 { provide: SessionStorageService, useClass: MockSyncStorage },
                 { provide: TranslateService, useClass: MockTranslateService },
+                { provide: textAssessmentService, useClass: TextAssessmentService },
+                { provide: programmingAssessmentService, useClass: ProgrammingAssessmentManualResultService },
+                { provide: fileUploadAssessmentService, useClass: FileUploadAssessmentService },
+                { provide: modelingAssessmentService, useClass: ModelingAssessmentService },
             ],
-        }).compileComponents();
+        })
+            .compileComponents()
+            .then(() => {
+                programmingAssessmentService = TestBed.inject(ProgrammingAssessmentManualResultService);
+                textAssessmentService = TestBed.inject(TextAssessmentService);
+                fileUploadAssessmentService = TestBed.inject(FileUploadAssessmentService);
+                modelingAssessmentService = TestBed.inject(ModelingAssessmentService);
+                courseService = TestBed.inject(CourseManagementService);
+            });
     });
 
     beforeEach(() => {
         fixture = TestBed.createComponent(AssessmentLocksComponent);
         component = fixture.componentInstance;
         fixture.detectChanges();
+        jest.resetAllMocks();
```

Comment:
```suggestion
        jest.restoreAllMocks();
```
Also consider moving this into an `afterEach()` call since this is the common way imo